github/git_refs.go

```diff
@@ -79,10 +79,12 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 	return r, resp, nil
 }
 
+// refURLEscape escapes every path segment of the given ref. Those must
+// not contain escaped "/" - as "%2F" - or github will not recognize it.
 func refURLEscape(ref string) string {
 	parts := strings.Split(ref, "/")
 	for i, s := range parts {
-		parts[i] = url.QueryEscape(s)
+		parts[i] = url.PathEscape(s)
```

Comment:
Hmmm... with this change from `url.QueryEscape` to `url.PathEscape`, I'm now concerned that we are breaking what was fixed in #1099.

Can you please add a unit test that demonstrates the URL found in #1099?

Unfortunately, I did not request that unit tests be added in #1099 to demonstrate the before/after changes. :-(
But I'm trying to fix that now.

Reply:
There's one in #1118 (issue #1101). This still works, `GetCommitSHA1()` got the `refURLEscape()` also. I've added a test case for the original trailing `%` as given in #1099 where http.NewRequest was failing to parse the URL.